src/gsi/setupgust.f90

```diff
@@ -395,33 +419,44 @@ subroutine setupgust(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,conv_diag
 
      if (zob > zges(1)) then
         factw=one
+        i_factw = 1
      else
         factw = data(iff10,i)
+        i_factw = 10
         if(sfcmod_gfs .or. sfcmod_mm5) then
            sfcr = data(isfcr,i)
            skint = data(iskint,i)
            call comp_fact10(dlat,dlon,dtime,skint,sfcr,isli,mype,factw)
+           i_factw = 5
         end if
 
         if (zob <= ten) then
            if(zob < ten)then
-              if (msonetob .and. twodvar_regional .and. use_similarity_2dvar) then
+              if (msonetob .and. (twodvar_regional .or. l_rtma3d) .and. use_similarity_2dvar) then
                  if (neutral_stability_windfact_2dvar) then
                     sfcr = data(isfcr,i)
                     factw=log(max(sfcr,zob)/sfcr)/log(ten/sfcr)
+                    i_factw = 3
```

Comment:
the additional i_factw will change behavior of the previous codes? or it is a fix missing in previous codes?

Reply:
@TingLei-NOAA 
Hi Ting,

This new variable, i_factw, does not change the behavior of the code, and it is also not a fix to the code. It actually has no any influence to the results of the code. I just used  i_factw to mark the way how "factw" was re-computed or adjusted, and output it in the netcdf-format obs-diag file.

It might cause some confusion, so I will remove it from the code.

Thank you for pointing this out!

-Gang